packages/xdl/src/detach/IosNSBundle.ts

```diff
@@ -419,15 +418,17 @@ async function _configureShellPlistAsync(context) {
     if (parseSdkMajorVersion(config.sdkVersion) < 28) {
       shellPlist.isShell = true;
     }
-    shellPlist.manifestUrl = context.published.url;
-    shellPlist.releaseChannel = context.published.releaseChannel;
-    if (context.data.testEnvironment) {
+    if (context.published) {
```

Comment:
Looks like the type of `StandaloneContext.published` is wrong: it's always defined in both `StandaloneContextUser` and `StandaloneContextService`, so this null check should be unnecessary.